### PR TITLE
Mark ongoing backup as failed

### LINKED ISSUE #10189: On leader change, new backup manager marks previous ongoing backup as failed
As specified in [ZEP](https://github.com/zeebe-io/enhancements/blob/master/ZEP008-hot-backup.md#backupactor), after replay is completed, backup manager should mark all on-going backup as failed. The new leader will not continue the backup initiated by the old leaders. To ensure that the users doesn't wait for ever for the backup to complete, the new leader marks them as failed.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

The new leader will not continue taking the backup which was initiated by the previous leader. Instead it marks it as failed, so that the users can monitor it and retry if required. We do not delete it so that the users can observe what happens. Users can delete the backup explicitly. 

Note:- If there are more than one backup in progress, it only mark the last one as failed. Ideally we should delete all previous checkpoints. But to keep it simple we just delete the previous one. We can improve this later, and can also provide a "prune" operation to delete old and incomplete backups.

## Related issues

closes #10189 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by lenaschoenburg — APPROVED** [review]
:rocket:

:wrench: We could add one more test to ensure that when `markFailed` fails for one backup, others are still marked as failed. 

:wrench: We could add one more test to ensure that exceptions are handled correctly and don't bubble up to the actor scheduler.

File: backup/src/main/java/io/camunda/zeebe/backup/processing/CheckpointRecordsProcessor.java [hunk 1]
```diff
@@ -126,4 +132,12 @@ public void addCheckpointListener(final CheckpointListener checkpointListener) {
           });
     }
   }
+
+  @Override
+  public void onRecovered(final ReadonlyStreamProcessorContext context) {
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply backup/src/main/java/io/camunda/zeebe/backup/processing/CheckpointRecordsProcessor.java:137]: I don't think that is necessary.

File: backup/src/main/java/io/camunda/zeebe/backup/management/BackupServiceImpl.java [hunk 2]
```diff
@@ -138,4 +145,48 @@ ActorFuture<BackupStatus> getBackupStatus(
...
+    backupStore
+        .getStatus(backupId)
+        .thenAccept(
+            status -> {
+              if (status.statusCode() == BackupStatusCode.IN_PROGRESS) {
+                LOG.debug(
+                    "The backup {} initiated by previous leader is still in progress. Marking it as failed.",
+                    backupId);
+                backupStore
+                    .markFailed(backupId)
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply backup/src/main/java/io/camunda/zeebe/backup/management/BackupServiceImpl.java:177]: Yes. I wanted to do it as part of this PR. But forgot to do it :woman_facepalming:

[the remark this answers is not in the document]
Comment (deepthidevaki) [reply backup/src/main/java/io/camunda/zeebe/backup/management/BackupServiceImpl.java]: @oleschoenburg I updated it https://github.com/camunda/zeebe/pull/10232/commits/d54a220ca113234441ad995a0a66767893da358d

**Review by lenaschoenburg — APPROVED** [review]
Thanks @deepthidevaki :+1: 

Just one rebase and this is done :tada:

**Comment by deepthidevaki** [comment]
bors merge

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**